src/controllers/user.js

```diff
@@ -0,0 +1,59 @@
+import Model from '../models';
+
+const { Users } = Model;
+
+/**
+* Generic update profile
+
+* @function
+*
+* @param {request} request - express request object
+*
+* @return {Object} - callback that execute the controller
+*
+* @description allows users to update their profile
+*/
+const updateProfile = async (request) => {
+  const {
+    params: { id },
+    body: payload
+  } = request;
+
+  const [numberOfAffectedRows, affectedRows] = await Users.update(
+    { ...payload },
+    { where: { id }, returning: true, raw: true }
+  );
+  if (numberOfAffectedRows === 0) return { status: 400, message: 'not updated' };
```

Comment:
Good job but I'm sure the user would like to know why it wasn't updated. Also, 400 seems like a weird error status for a failed update that doesn't seem to be the fault of the user.

Reply:
Done.